src/controllers/comment.controller/removeComment/index.js

```diff
@@ -0,0 +1,57 @@
+const logger = require('../../../config/logger.js');
+const { getUserId } = require('./queries/queries.js');
+const { removeProjectComment, removeIdeaComment, removeThoughtComment } = require('./queries/mutations.js');
+const { query: Hasura } = require('../../../utils/hasura.js');
+const catchAsync = require('../../../utils/catchAsync.js');
+
+const handleRemoveComment = async (updateFunction, articleType, commentId, userId, comment) => {
+  const updateResponse = await Hasura(updateFunction, { id: commentId, userId, text: comment });
+  if (!updateResponse.success) {
+    return {
+      success: false,
+      errorCode: updateResponse.errorCode,
+      errorMessage: updateResponse.errorMessage,
+      data: null,
+    };
+  }
+
+  return {
+    success: true,
+    errorCode: '',
+    errorMessage: '',
+    data: null,
+  };
+};
+
+const removeComment = catchAsync(async (req, res) => {
+  const { commentId } = req.params;
+  const { cognito_sub, articleType } = req.body;
+
+  const getUserIdResponse = await Hasura(getUserId, { cognito_sub: { _eq: cognito_sub } });
```

Comment:
No need the fetch the user id here, you can directly delete the comment based on the cognito_sub. Check comment on mutations